test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -831,12 +832,12 @@ private static void assertListEquals(List<Object> expected, List<Object> actual)
      * for byte[] through Arrays.equals(byte[], byte[])
      */
     @SuppressWarnings("unchecked")
-    private static void assertObjectEquals(Object expected, Object actual) {
+    private static void assertObjectEquals(Object expected, Object actual, String path) {
         if (expected instanceof Map) {
             assertThat(actual, instanceOf(Map.class));
-            assertMapEquals((Map<String, Object>) expected, (Map<String, Object>) actual);
+            assertMapEquals((Map<String, Object>) expected, (Map<String, Object>) actual, path);
         } else if (expected instanceof List) {
-            assertListEquals((List<Object>) expected, (List<Object>) actual);
+            assertListEquals((List<Object>) expected, (List<Object>) actual, path);
```

Comment:
I added these path arguments to be able to better debug errors in map-comparisons. When we compare xContent and some nested map has different size, we currently only get the expected and actual size as output. This is very useful to print in the test error message, so I'd like to leave it.